### PR TITLE
chore: Add versions mdx

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary

Fixes: https://github.com/Workday/canvas-kit/issues/2521

## Release Category
Documentation

### Release Note
We've added a table of previous versions to storybook

---

## Checklist

- [ ] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [ ] Label `ready for review` has been added to PR

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation
- [ ] Testing
- [ ] Codemods

### CODE REVIEW

File: modules/docs/mdx/Versions.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import {cssVar} from '@workday/canvas-kit-styling';
+import {version} from '../../../lerna.json';
+const allVersions = [
+  {
+    versionNumber: version,
+    documentation: 'https://github.com/Workday/canvas-kit',
+  },
+  {
+    versionNumber: 9,
+    documentation: 'https://d2krrudi3mmzzw.cloudfront.net/v9/?path=/docs/welcome--page',
```
Comment (alanbsmith) [inline modules/docs/mdx/Versions.mdx:13]: Are these urls stable?

File: modules/docs/mdx/Versions.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      <Table.Header scope="col">Version</Table.Header>
+      <Table.Header scope="col">Documentation</Table.Header>
+      <Table.Header scope="col">Status</Table.Header>
+    </Table.Row>
+  </Table.Head>
+  <Table.Body>
+    {allVersions.map(item => {
+      let chooseVariant;
+      let status;
+      if (item.versionNumber === version) {
```
Comment (alanbsmith) [inline modules/docs/mdx/Versions.mdx:49]: Question: Why don't you need to parse the version string here?

File: modules/docs/mdx/Versions.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    documentation:
+      'https://d2krrudi3mmzzw.cloudfront.net/v7/?path=/story/welcome-getting-started--page',
+  },
+  {
+    versionNumber: 6,
+    documentation:
+      'https://d2krrudi3mmzzw.cloudfront.net/v7/?path=/story/welcome-getting-started--page',
+  },
+];
+
```
Comment (alanbsmith) [inline modules/docs/mdx/Versions.mdx:30]:
```suggestion
const statusIndicators = {
  stable: {
    variant: 'blue',
    label: 'Stable',
  },
    support: {
    variant: 'green',
    label: 'Support',
  },
    deprecated: {
    variant: 'red',
    label: 'No Longer Supported',
  }
};

function getVersionStatusIndicator(versionNumber: number) {
  const currentMajorVersion = Number(version.split('.')[0]);
  // if this is the current version
  if (vesionNumber === currentMajorVersion) {
    return statusIndicators.stable;
  } else if (versionNumber === currentMajorVersion - 1) {
    return statusIndicators.support;
  } else {
    return statusIndicators.deprecated;
  }
}
```

File: modules/docs/mdx/Versions.mdx (hunk 2 above)
Comment (alanbsmith) [reply modules/docs/mdx/Versions.mdx:49]: `version` should be a string (I think), and `versionNumber` is a number. they'll never be equal.

File: modules/docs/mdx/Versions.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        chooseVariant = 'blue';
+        status = 'Stable';
+      } else if (item.versionNumber === version.split('.')[0] - 1) {
+        chooseVariant = 'green';
+        status = 'Support';
+      } else {
+        chooseVariant = 'red';
+        status = 'No Longer Supported';
+      }
+      return (
```
Comment (alanbsmith) [inline modules/docs/mdx/Versions.mdx:59]: Comment: I appreciate you trying to keep this all in one spot, but I think it's more readable if you split it out.

File: modules/docs/mdx/Versions.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      if (item.versionNumber === version) {
+        chooseVariant = 'blue';
+        status = 'Stable';
+      } else if (item.versionNumber === version.split('.')[0] - 1) {
+        chooseVariant = 'green';
+        status = 'Support';
+      } else {
+        chooseVariant = 'red';
+        status = 'No Longer Supported';
+      }
```
Comment (alanbsmith) [inline modules/docs/mdx/Versions.mdx:58]:
```suggestion
    {allVersions.map(item => {
      const {label, variant} = getVersionStatusIndicator(item.version);
```

File: modules/docs/mdx/Versions.mdx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        <Table.Row>
+          <Table.Cell>v{item.versionNumber}</Table.Cell>
+          <Table.Cell>
+            <a href={item.documentation} target="_blank">
+              Documentation
+            </a>
+          </Table.Cell>
+          <Table.Cell>
+            <StatusIndicator variant={chooseVariant}>
+              <StatusIndicator.Label>{status}</StatusIndicator.Label>
```
Comment (alanbsmith) [inline modules/docs/mdx/Versions.mdx:69]:
```suggestion
            <StatusIndicator variant={variant.}>
              <StatusIndicator.Label>{label}</StatusIndicator.Label>
```

**Review by alanbsmith — CHANGES_REQUESTED** [review]
Thanks for working on this. I suggested some changes. 😄

File: modules/docs/mdx/Versions.mdx (hunk 1 above)
Comment (mannycarrera4) [reply modules/docs/mdx/Versions.mdx:13]: yeah

File: modules/docs/mdx/versionsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    label: 'Support',
+  },
+  deprecated: {
+    variant: 'red',
+    label: 'No Longer Supported',
+  },
+};
+function getVersionStatusIndicator(versionNumber: number | string) {
+  const currentMajorVersion = Number(version.split('.')[0]);
+  // version from lerna is a string, so we need to do modify into a number
```
Comment (alanbsmith) [inline modules/docs/mdx/versionsTable.tsx:48]: This comment should be above line 47. `version` is the version from `lerna.json`.

File: modules/docs/mdx/versionsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        </Table.Row>
+      </Table.Head>
+      <Table.Body>
+        {allVersions.map(item => {
+          const {label, variant} = getVersionStatusIndicator(item.versionNumber);
+          return (
+            <Table.Row>
+              <Table.Cell>v{item.versionNumber}</Table.Cell>
+              <Table.Cell>
+                <a href={item.documentation} target="_blank">
```
Comment (alanbsmith) [inline modules/docs/mdx/versionsTable.tsx:79]:
```suggestion
                <a href={item.documentation} target="_blank" rel="noreferrer">
```

**Review by alanbsmith — APPROVED** [review]

File: modules/docs/mdx/versionsTable.tsx (hunk 7 above)
Comment (mannycarrera4) [inline modules/docs/mdx/versionsTable.tsx:48]:
```suggestion
```

File: modules/docs/mdx/versionsTable.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    variant: 'green',
+    label: 'Support',
+  },
+  deprecated: {
+    variant: 'red',
+    label: 'No Longer Supported',
+  },
+};
+function getVersionStatusIndicator(versionNumber: number | string) {
+  const currentMajorVersion = Number(version.split('.')[0]);
```
Comment (mannycarrera4) [inline modules/docs/mdx/versionsTable.tsx:47]:
```suggestion
  // version from lerna is a string, so we need to do modify into a number
  const currentMajorVersion = Number(version.split('.')[0]);
```

**Pull request merged**